coral-hive/src/main/java/com/linkedin/coral/transformers/CoralRelNodeToCoralSqlNodeTransformer.java

```diff
@@ -0,0 +1,172 @@
+ * See LICENSE in the project root for license information.
+ */
+package com.linkedin.coral.transformers;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.ConcurrentHashMap;
+
+import org.apache.calcite.config.NullCollation;
+import org.apache.calcite.linq4j.tree.Expressions;
+import org.apache.calcite.rel.core.Correlate;
+import org.apache.calcite.rel.core.Project;
+import org.apache.calcite.rel.core.TableScan;
+import org.apache.calcite.rel.core.Uncollect;
+import org.apache.calcite.rel.rel2sql.RelToSqlConverter;
+import org.apache.calcite.rel.type.RelDataType;
+import org.apache.calcite.rex.RexNode;
+import org.apache.calcite.sql.JoinConditionType;
+import org.apache.calcite.sql.JoinType;
+import org.apache.calcite.sql.SqlBasicCall;
+import org.apache.calcite.sql.SqlDialect;
+import org.apache.calcite.sql.SqlIdentifier;
+import org.apache.calcite.sql.SqlJoin;
+import org.apache.calcite.sql.SqlLiteral;
+import org.apache.calcite.sql.SqlNode;
+import org.apache.calcite.sql.fun.SqlStdOperatorTable;
+import org.apache.calcite.sql.parser.SqlParserPos;
+import org.apache.calcite.util.Util;
+
+import com.linkedin.coral.com.google.common.collect.ImmutableList;
+import com.linkedin.coral.com.google.common.collect.ImmutableMap;
+import com.linkedin.coral.common.functions.Function;
+import com.linkedin.coral.hive.hive2rel.functions.HiveFunctionResolver;
+import com.linkedin.coral.hive.hive2rel.functions.StaticHiveFunctionRegistry;
+
+
+/**
+ * This class converts Coral RelNode to Coral SqlNode.
+ *
+ * This class currently handles only lateral view transformations
+ */
+public class CoralRelNodeToCoralSqlNodeTransformer extends RelToSqlConverter {
+
+  public static final SqlDialect INSTANCE = returnInstance();
+  private final HiveFunctionResolver functionResolver =
+      new HiveFunctionResolver(new StaticHiveFunctionRegistry(), new ConcurrentHashMap<>());
+
+  /**
+   * Creates a CoralRelNodeToCoralSqlNodeTransformer.
+   */
+  CoralRelNodeToCoralSqlNodeTransformer() {
+    super(INSTANCE);
+  }
+
+  private static SqlDialect returnInstance() {
+    SqlDialect.Context context = SqlDialect.EMPTY_CONTEXT.withDatabaseProduct(SqlDialect.DatabaseProduct.HIVE)
+        .withNullCollation(NullCollation.HIGH);
+
+    return new SqlDialect(context);
+  }
+
+  /**
+   * This overridden function makes sure that the basetable names in the output SqlNode
+   * will be in the format - "dbname.tablename" instead of "catalogname.dbname.tablename"
+   * For Example:
+   * hive.default.complex will be converted to default.complex
+   *
+   * input RelNode format is:
+   * LogicalTableScan(table=[[hive, default, complex]])
+   * This overriding function creates a SqlIdentifier(default, complex)
+   * instead of super's default: SqlIdentifier(hive, default, complex)
+   */
+  @Override
+  public Result visit(TableScan e) {
+    //todo: look into removing this transformation during hiveToRelConversion
```

Comment:
nit: // TODO:

Reply:
Good point. I removed the todo from here and filed the issue https://github.com/linkedin/coral/issues/284.